plugins/modules/vmware_guest.py

```diff
@@ -2586,6 +2639,35 @@ def configure_disks(self, vm_obj):
 
                 self.change_detected = True
 
+    def create_directory_for_datadisk(self, datacenter, datastore, vm_name, diskspec):
+        """
+        Helper function to create a directory for data disk
+        """
+        parent_dc = datacenter
+        datastore_name = datastore.name
+        # Try to create folder for virtual machine and disks inside datastore
+        path_on_ds = '[' + datastore_name + ']' + vm_name
+        try:
+            # patch on top of patch: This directory creation is not needed because VMware creates the directories itself
+            # When it is there, it causes extra incremented _1 directories to be created by VMware. (may only apply to 6.7, not 7.0)
+            pass
```

Comment:
`pass` in try-except is useless.

Reply:
need to re-test this on ESXi 7, may need a combined fix.

Reply:
Per my last comment in #353,

This is what I have now:
```
            if not self.is_vcenter():
                self.content.fileManager.MakeDirectory(name=path_on_ds,
                                                       datacenter=parent_dc,
                                                       createParentDirectories=True)
```
but do need to test more version combinations. My test environment is vcenter 7 with ESXi 6.7. It may be different with ESXi 7.